### ISSUE TITLE
wsl : The term 'wsl' is not recognized as the name of a cmdlet,

### ISSUE THREAD

**Issue by geeksupport1** [MicrosoftDocs/WSL#499]
Hi,

 I cannot run wsl.

 I have install ubuntu18.02 through windows store and running it.

I am using Window 10 home on a Corei7 4720HQ

OS build 18995.1

My windows version is
`Microsoft Windows [Version 10.0.18995.1]`

I have followed instructions on
https://docs.microsoft.com/en-us/windows/wsl/wsl2-install

When in powershell as admin i type in
```
PS C:\WINDOWS\system32> wsl
wsl : The term 'wsl' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ wsl
+ ~~~
    + CategoryInfo          : ObjectNotFound: (wsl:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

```
I cannot see wsl in the path
`dir C:\Windows\System32\wsl*`

Any hints?

I have tried all the commands in the links.
https://docs.microsoft.com/en-us/windows/wsl/wsl2-install
https://github.com/MicrosoftDocs/WSL/issues/405

Thanks in advance.

---

#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b19a43be-b16a-d903-b69d-75dffce6c329
* Version Independent ID: f9d4a118-1251-8d6b-43f5-2e64808dc527
* Content: [Install WSL 2](https://docs.microsoft.com/en-us/windows/wsl/wsl2-install)
* Content Source: [WSL/wsl2-install.md](https://github.com/MicrosoftDocs/WSL/blob/live/WSL/wsl2-install.md)
* Product: **dev-environment**
* Technology: **windows-subsystem-for-linux**
* GitHub Login: @craigloewen-msft
* Microsoft Alias: **crloewen**

**Comment by geeksupport1** [comment]
Strange as it is.

wsl -l v works in CMD but not powershell.

So issued is resolved.

wsl is visible in cmd.exe but not in powershell.

Do not know why.

The rest of the commands relating to wsl wslconfig work in cmd but not powershell.

I have restarted the PC many times.

regards

**Comment by plassa** [comment]
I have same issue where wsl works invoked from cmd.exe, but not from powershell

@geeksupport1, why did you close this?

PS C:\WINDOWS\system32> wsl
wsl : The term 'wsl' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ wsl
+ ~~~
    + CategoryInfo          : ObjectNotFound: (wsl:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS C:\WINDOWS\system32>

**Comment by timgilbert** [comment]
I was seeing the same behavior in Windows 10.0.19025.1, but after I installed an Ubuntu distribution from the MS Store, the `wsl` command started working for me in both Powershell and `cmd.exe`.

**Comment by tkr789** [comment]
I have done all of the above in Windows [Version 10.0.19041.172]. Reinstalled Ubuntu 18.04. In **powershell** using `> wsl -l`:

wsl : The term 'wsl' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ wsl -l
+ ~~~
    + CategoryInfo          : ObjectNotFound: (wsl:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Using `> wsl --set-version <Ubuntu-18.04> 2`:
At line:1 char:19
+ wsl --set-version <Ubuntu-18.04> 2
+                   ~
The '<' operator is reserved for future use.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : RedirectionNotSupported

In **cmd**:

  `> wsl -l`:
  NAME            STATE           VERSION
* Ubuntu-18.04    Stopped         1

`wsl --set-version <Ubuntu-18.04> 2`:
The system cannot find the file specified.

**Comment by JayElOpia** [comment]
@geeksupport1 
If this command : 

Get-WindowsOptionalFeature -Online | Where-Object {$_.FeatureName -eq "Microsoft-Windows-Subsystem-Linux"} | Format-Table

shows you the, state as  'Enabled'  while running powershell as Admin, try using the 64Bit version of Powershell (powershell 6) instead. That worked for me.

**Comment by JayElOpia** [comment]
@geeksupport1  For this command :  wsl --set-version <Ubuntu-18.04> 2  the angle brackets shouldn't be around the OS name.

**Comment by octy40** [comment]
@JayElOpia - 64bit version worked. Thank you!

**Comment by cheater** [comment]
After installing the latest version of powershell (7.1.0-preview.7 - 2020-09-08, 64-bit version) wsl is still not being found.

**Comment by raajpackt** [comment]
same issue here.

**Comment by TeunvdWeij** [comment]
Running this: 'dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart' 

Fixed my problem. You need to install wsl first for it to be recognized. To be honest, the docs are quite vague but now it seems to work for me. 

I got the command from https://docs.microsoft.com/en-us/windows/wsl/install-win10 in the first manual installation section
.

**Comment by remco-pc** [comment]
> Strange as it is.
> 
> wsl -l v works in CMD but not powershell.
> 
> So issued is resolved.
> 
> wsl is visible in cmd.exe but not in powershell.
> 
> Do not know why.
> 
> The rest of the commands relating to wsl wslconfig work in cmd but not powershell.
> 
> I have restarted the PC many times.
> 
> regards

You need to install WSL2 Linux kernel update package for x64 machines. then it works like a charm

**Comment by yudai-patronai** [comment]
> 
> 
> @tkr789
> If this command :
> 
> Get-WindowsOptionalFeature -Online | Where-Object {$_.FeatureName -eq "Microsoft-Windows-Subsystem-Linux"} | Format-Table
> 
> shows you the, state as 'Enabled' while running powershell as Admin, try using the 64Bit version of Powershell (powershell 6) instead. That worked for me.

It shows Enabled , but neither of powershells allows to install update wsl for me

**Comment by AryanBhandari47** [comment]
> I have done all of the above in Windows [Version 10.0.19041.172]. Reinstalled Ubuntu 18.04. In **powershell**:
> 
> `> wsl -l`:
> wsl : The term 'wsl' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
> spelling of the name, or if a path was included, verify that the path is correct and try again.
> 
> `> wsl --set-version <Ubuntu-18.04> 2`:
> At line:1 char:19
> wsl --set-version <Ubuntu-18.04> 2
> The '<' operator is reserved for future use.
> + CategoryInfo : ParserError: (:) [], ParentContainsErrorRecordException
> + FullyQualifiedErrorId : RedirectionNotSupported
> 
> In **cmd**:
> 
> `> wsl -l`:
> NAME STATE VERSION
> Ubuntu-18.04 Stopped 1
> 
> `wsl --set-version <Ubuntu-18.04> 2`:
> The system cannot find the file specified.
> 
> I have not found anything that fixes this.


First make sure your turn on the following windows features :
- **Windows hypervisor platform**
- **Windows subsystem for linux**
**Now .**
1) Run cmd with admin
2) Run wsl --install
3) Run wsl --set-default-version 2 (this will set your wsl default to 2 but not ubuntu's)
4) Run wsl --set-version Ubuntu-18.04 2
5) Restart or refresh your docker desktop , go to settings>resources>wsl integration. 
You can now enable integration. 
**NOTE : wsl still won't work with PowerShell**

**Comment by chicacode** [comment]
i had the same issue,
and restart the computer works for me

**Comment by dianavintila** [comment]
> Run the powershell as administrator then run this command:
> 
> ```
> dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart
> ```
> 
> Additionally, in my case, I added System32 to the `PATH` environment variables. I hope this helps someone. :) ![image](https://user-images.githubusercontent.com/10904957/205212754-d9f8a2fd-fb70-4750-b100-a4fa55c7f1a8.png)
It worked for me :)

**Comment by jasurmavlonov02** [comment]
There is no standard opinion

**Comment by garryyan2** [comment]
For my case, wsl was not recognized in PowerShell (It had worked originally). wsl is recognized in Command Prompt. I reckoned that the issue is PoweShell's "PATH" variable is not set properly. It works after I added "C:\Windows\sysnative" directory to the Path in "Environment" in HKEY_CURRENT_USER by running Eegedit and restart the PC. This might not be the correct way to do it. It just works for me.
